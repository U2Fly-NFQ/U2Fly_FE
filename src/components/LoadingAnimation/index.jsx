const loadingStyle = {
  width: '100%',
  objectFit: 'contain',
}

const LoadingAnimation = () => {
  return <img style={loadingStyle} alt="Loading animation" />
}

export default LoadingAnimation
