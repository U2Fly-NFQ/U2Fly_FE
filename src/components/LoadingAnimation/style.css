@charset "UTF-8";

.loading-animation {
  display: flex;
  justify-content: center;
  align-items: center;
  background-color: transparent;
  position: relative;
  z-index: 10;
  height: 200px;
  width: 200px;
  max-width: 100%;
}
.loading-animation .loader {
  padding: 30px 0;
  width: 150px;
  height: 150px;
  border: 1px solid rgba(125, 125, 125, 0.15);
  border-radius: 100%;
}
.loading-animation .hide {
  display: none;
}
.loading-animation .loader2 {
  position: relative;
  width: 120px;
  height: 120px;
}
.loading-animation .loader2 span {
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  transform: rotate(calc(18deg * var(--i)));
}
.loading-animation .loader2 span:before {
  content: '';
  position: absolute;
  top: 0;
  left: 0;
  width: 15px;
  height: 15px;
  background: var(--ant-primary-color);
  border-radius: 50%;
  transform: scale(0);
  animation: animate 1s linear infinite;
  animation-delay: calc(0.05s * var(--i));
}
.loading-animation .loader2 .plane {
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  animation: rotating 1s linear infinite;
  animation-delay: -0.5s;
}
.loading-animation .loader2 .plane:before {
  content: '\f072';
  font-family: fontAwesome;
  position: absolute;
  top: 70px;
  left: 75px;
  color: var(--ant-primary-color);
  font-size: 55px;
  transform: rotate(140deg);
}

@keyframes animate {
  0% {
    transform: scale(0);
  }
  50% {
    transform: scale(1.2);
  }
  80%,
  100% {
    transform: scale(0);
  }
}
@keyframes rotating {
  0% {
    transform: rotate(10deg);
  }
  100% {
    transform: rotate(370deg);
  }
}
