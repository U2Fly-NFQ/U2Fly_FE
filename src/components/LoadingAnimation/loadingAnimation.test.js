import renderer from 'react-test-renderer'
import LoadingAnimation from './'

it('renders correctly', () => {
  const tree = renderer.create(<LoadingAnimation />).toJSON()
  expect(tree).toMatchSnapshot()
})
