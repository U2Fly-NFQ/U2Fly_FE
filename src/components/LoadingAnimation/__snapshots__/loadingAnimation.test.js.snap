// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<img
  alt="Loading animation"
  style={
    Object {
      "maxWidth": "100%",
      "objectFit": "contain",
    }
  }
/>
`;
